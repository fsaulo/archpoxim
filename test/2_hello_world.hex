0xDC00000E
0x00000000
0x00000000
0x00000000
0x00000000
0x00000000
0x00000000
0x00000000
0x60620000
0x5C030000
0xB8000003
0x6C610000
0x48420001
0xDFFFFFFA
0x7C000000
0x03C07FFC
0x6820001B
0x00400050
0xE7FFFFF5
0xFC000000
0x0A48656C
0x6C6F2077
0x6F726C64
0x2066726F
0x6D20506F
0x78696D21
0x0A000000
0x8888888B