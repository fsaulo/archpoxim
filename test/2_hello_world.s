// Segmento de codigo
.text
    // Tabela de vetor de interrupcao
    init:
        bun main
        .align 5
    // Procedimento de impressao
    printf:
        // Leitura de byte da mensagem
        l8 r3, [r2]
        // Comparando com '\0'
        cmpi r3, 0
        // Finalizando se for caractere nulo
        beq 3
        // Escreve no terminal
        s8 [r1], r3
        // Incrementa o ponteiro da mensagem
        addi r2, r2, 1
        // Repete a iteracao
        bun -6
        // Retorno da funcao
        ret
    // Funcao principal
    main:
        // SP = 32 KiB
        mov sp, 0x7FFC
        // R1 = endereco do terminal
        l32 r1, [terminal]
        // R2 = ponteiro da string
        mov r2, mensagem
        // printf
        call printf
        // Finalizacao de execucao
        int 0
// Segmento de dados
.data
    // Mensagem de texto
    mensagem:
        .asciz "\nHello world from Poxim!\n"
    // Endereco do dispositivo (OUT)
    terminal:
        .4byte 0x8888888B