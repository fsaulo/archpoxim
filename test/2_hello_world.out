[START OF SIMULATION]
0x00000000:	bun 14                   	PC=0x0000003C
0x0000003C:	mov sp,32764             	SP=0x00007FFC
0x00000040:	l32 r1,[r0+27]           	R1=MEM[0x0000006C]=0x8888888B
0x00000044:	mov r2,80                	R2=0x00000050
0x00000048:	call -11                 	PC=0x00000020,MEM[0x00007FFC]=0x0000004C
0x00000020:	l8 r3,[r2+0]             	R3=MEM[0x00000050]=0x0A
0x00000024:	cmpi r3,0                	SR=0x00000000
0x00000028:	beq 3                    	PC=0x0000002C
0x0000002C:	s8 [r1+0],r3             	MEM[0x8888888B]=R3=0x0A
0x00000030:	addi r2,r2,1             	R2=R2+0x00000001=0x00000051,SR=0x00000000
0x00000034:	bun -6                   	PC=0x00000020
0x00000020:	l8 r3,[r2+0]             	R3=MEM[0x00000051]=0x48
0x00000024:	cmpi r3,0                	SR=0x00000000
0x00000028:	beq 3                    	PC=0x0000002C
0x0000002C:	s8 [r1+0],r3             	MEM[0x8888888B]=R3=0x48
0x00000030:	addi r2,r2,1             	R2=R2+0x00000001=0x00000052,SR=0x00000000
0x00000034:	bun -6                   	PC=0x00000020
0x00000020:	l8 r3,[r2+0]             	R3=MEM[0x00000052]=0x65
0x00000024:	cmpi r3,0                	SR=0x00000000
0x00000028:	beq 3                    	PC=0x0000002C
0x0000002C:	s8 [r1+0],r3             	MEM[0x8888888B]=R3=0x65
0x00000030:	addi r2,r2,1             	R2=R2+0x00000001=0x00000053,SR=0x00000000
0x00000034:	bun -6                   	PC=0x00000020
0x00000020:	l8 r3,[r2+0]             	R3=MEM[0x00000053]=0x6C
0x00000024:	cmpi r3,0                	SR=0x00000000
0x00000028:	beq 3                    	PC=0x0000002C
0x0000002C:	s8 [r1+0],r3             	MEM[0x8888888B]=R3=0x6C
0x00000030:	addi r2,r2,1             	R2=R2+0x00000001=0x00000054,SR=0x00000000
0x00000034:	bun -6                   	PC=0x00000020
0x00000020:	l8 r3,[r2+0]             	R3=MEM[0x00000054]=0x6C
0x00000024:	cmpi r3,0                	SR=0x00000000
0x00000028:	beq 3                    	PC=0x0000002C
0x0000002C:	s8 [r1+0],r3             	MEM[0x8888888B]=R3=0x6C
0x00000030:	addi r2,r2,1             	R2=R2+0x00000001=0x00000055,SR=0x00000000
0x00000034:	bun -6                   	PC=0x00000020
0x00000020:	l8 r3,[r2+0]             	R3=MEM[0x00000055]=0x6F
0x00000024:	cmpi r3,0                	SR=0x00000000
0x00000028:	beq 3                    	PC=0x0000002C
0x0000002C:	s8 [r1+0],r3             	MEM[0x8888888B]=R3=0x6F
0x00000030:	addi r2,r2,1             	R2=R2+0x00000001=0x00000056,SR=0x00000000
0x00000034:	bun -6                   	PC=0x00000020
0x00000020:	l8 r3,[r2+0]             	R3=MEM[0x00000056]=0x20
0x00000024:	cmpi r3,0                	SR=0x00000000
0x00000028:	beq 3                    	PC=0x0000002C
0x0000002C:	s8 [r1+0],r3             	MEM[0x8888888B]=R3=0x20
0x00000030:	addi r2,r2,1             	R2=R2+0x00000001=0x00000057,SR=0x00000000
0x00000034:	bun -6                   	PC=0x00000020
0x00000020:	l8 r3,[r2+0]             	R3=MEM[0x00000057]=0x77
0x00000024:	cmpi r3,0                	SR=0x00000000
0x00000028:	beq 3                    	PC=0x0000002C
0x0000002C:	s8 [r1+0],r3             	MEM[0x8888888B]=R3=0x77
0x00000030:	addi r2,r2,1             	R2=R2+0x00000001=0x00000058,SR=0x00000000
0x00000034:	bun -6                   	PC=0x00000020
0x00000020:	l8 r3,[r2+0]             	R3=MEM[0x00000058]=0x6F
0x00000024:	cmpi r3,0                	SR=0x00000000
0x00000028:	beq 3                    	PC=0x0000002C
0x0000002C:	s8 [r1+0],r3             	MEM[0x8888888B]=R3=0x6F
0x00000030:	addi r2,r2,1             	R2=R2+0x00000001=0x00000059,SR=0x00000000
0x00000034:	bun -6                   	PC=0x00000020
0x00000020:	l8 r3,[r2+0]             	R3=MEM[0x00000059]=0x72
0x00000024:	cmpi r3,0                	SR=0x00000000
0x00000028:	beq 3                    	PC=0x0000002C
0x0000002C:	s8 [r1+0],r3             	MEM[0x8888888B]=R3=0x72
0x00000030:	addi r2,r2,1             	R2=R2+0x00000001=0x0000005A,SR=0x00000000
0x00000034:	bun -6                   	PC=0x00000020
0x00000020:	l8 r3,[r2+0]             	R3=MEM[0x0000005A]=0x6C
0x00000024:	cmpi r3,0                	SR=0x00000000
0x00000028:	beq 3                    	PC=0x0000002C
0x0000002C:	s8 [r1+0],r3             	MEM[0x8888888B]=R3=0x6C
0x00000030:	addi r2,r2,1             	R2=R2+0x00000001=0x0000005B,SR=0x00000000
0x00000034:	bun -6                   	PC=0x00000020
0x00000020:	l8 r3,[r2+0]             	R3=MEM[0x0000005B]=0x64
0x00000024:	cmpi r3,0                	SR=0x00000000
0x00000028:	beq 3                    	PC=0x0000002C
0x0000002C:	s8 [r1+0],r3             	MEM[0x8888888B]=R3=0x64
0x00000030:	addi r2,r2,1             	R2=R2+0x00000001=0x0000005C,SR=0x00000000
0x00000034:	bun -6                   	PC=0x00000020
0x00000020:	l8 r3,[r2+0]             	R3=MEM[0x0000005C]=0x20
0x00000024:	cmpi r3,0                	SR=0x00000000
0x00000028:	beq 3                    	PC=0x0000002C
0x0000002C:	s8 [r1+0],r3             	MEM[0x8888888B]=R3=0x20
0x00000030:	addi r2,r2,1             	R2=R2+0x00000001=0x0000005D,SR=0x00000000
0x00000034:	bun -6                   	PC=0x00000020
0x00000020:	l8 r3,[r2+0]             	R3=MEM[0x0000005D]=0x66
0x00000024:	cmpi r3,0                	SR=0x00000000
0x00000028:	beq 3                    	PC=0x0000002C
0x0000002C:	s8 [r1+0],r3             	MEM[0x8888888B]=R3=0x66
0x00000030:	addi r2,r2,1             	R2=R2+0x00000001=0x0000005E,SR=0x00000000
0x00000034:	bun -6                   	PC=0x00000020
0x00000020:	l8 r3,[r2+0]             	R3=MEM[0x0000005E]=0x72
0x00000024:	cmpi r3,0                	SR=0x00000000
0x00000028:	beq 3                    	PC=0x0000002C
0x0000002C:	s8 [r1+0],r3             	MEM[0x8888888B]=R3=0x72
0x00000030:	addi r2,r2,1             	R2=R2+0x00000001=0x0000005F,SR=0x00000000
0x00000034:	bun -6                   	PC=0x00000020
0x00000020:	l8 r3,[r2+0]             	R3=MEM[0x0000005F]=0x6F
0x00000024:	cmpi r3,0                	SR=0x00000000
0x00000028:	beq 3                    	PC=0x0000002C
0x0000002C:	s8 [r1+0],r3             	MEM[0x8888888B]=R3=0x6F
0x00000030:	addi r2,r2,1             	R2=R2+0x00000001=0x00000060,SR=0x00000000
0x00000034:	bun -6                   	PC=0x00000020
0x00000020:	l8 r3,[r2+0]             	R3=MEM[0x00000060]=0x6D
0x00000024:	cmpi r3,0                	SR=0x00000000
0x00000028:	beq 3                    	PC=0x0000002C
0x0000002C:	s8 [r1+0],r3             	MEM[0x8888888B]=R3=0x6D
0x00000030:	addi r2,r2,1             	R2=R2+0x00000001=0x00000061,SR=0x00000000
0x00000034:	bun -6                   	PC=0x00000020
0x00000020:	l8 r3,[r2+0]             	R3=MEM[0x00000061]=0x20
0x00000024:	cmpi r3,0                	SR=0x00000000
0x00000028:	beq 3                    	PC=0x0000002C
0x0000002C:	s8 [r1+0],r3             	MEM[0x8888888B]=R3=0x20
0x00000030:	addi r2,r2,1             	R2=R2+0x00000001=0x00000062,SR=0x00000000
0x00000034:	bun -6                   	PC=0x00000020
0x00000020:	l8 r3,[r2+0]             	R3=MEM[0x00000062]=0x50
0x00000024:	cmpi r3,0                	SR=0x00000000
0x00000028:	beq 3                    	PC=0x0000002C
0x0000002C:	s8 [r1+0],r3             	MEM[0x8888888B]=R3=0x50
0x00000030:	addi r2,r2,1             	R2=R2+0x00000001=0x00000063,SR=0x00000000
0x00000034:	bun -6                   	PC=0x00000020
0x00000020:	l8 r3,[r2+0]             	R3=MEM[0x00000063]=0x6F
0x00000024:	cmpi r3,0                	SR=0x00000000
0x00000028:	beq 3                    	PC=0x0000002C
0x0000002C:	s8 [r1+0],r3             	MEM[0x8888888B]=R3=0x6F
0x00000030:	addi r2,r2,1             	R2=R2+0x00000001=0x00000064,SR=0x00000000
0x00000034:	bun -6                   	PC=0x00000020
0x00000020:	l8 r3,[r2+0]             	R3=MEM[0x00000064]=0x78
0x00000024:	cmpi r3,0                	SR=0x00000000
0x00000028:	beq 3                    	PC=0x0000002C
0x0000002C:	s8 [r1+0],r3             	MEM[0x8888888B]=R3=0x78
0x00000030:	addi r2,r2,1             	R2=R2+0x00000001=0x00000065,SR=0x00000000
0x00000034:	bun -6                   	PC=0x00000020
0x00000020:	l8 r3,[r2+0]             	R3=MEM[0x00000065]=0x69
0x00000024:	cmpi r3,0                	SR=0x00000000
0x00000028:	beq 3                    	PC=0x0000002C
0x0000002C:	s8 [r1+0],r3             	MEM[0x8888888B]=R3=0x69
0x00000030:	addi r2,r2,1             	R2=R2+0x00000001=0x00000066,SR=0x00000000
0x00000034:	bun -6                   	PC=0x00000020
0x00000020:	l8 r3,[r2+0]             	R3=MEM[0x00000066]=0x6D
0x00000024:	cmpi r3,0                	SR=0x00000000
0x00000028:	beq 3                    	PC=0x0000002C
0x0000002C:	s8 [r1+0],r3             	MEM[0x8888888B]=R3=0x6D
0x00000030:	addi r2,r2,1             	R2=R2+0x00000001=0x00000067,SR=0x00000000
0x00000034:	bun -6                   	PC=0x00000020
0x00000020:	l8 r3,[r2+0]             	R3=MEM[0x00000067]=0x21
0x00000024:	cmpi r3,0                	SR=0x00000000
0x00000028:	beq 3                    	PC=0x0000002C
0x0000002C:	s8 [r1+0],r3             	MEM[0x8888888B]=R3=0x21
0x00000030:	addi r2,r2,1             	R2=R2+0x00000001=0x00000068,SR=0x00000000
0x00000034:	bun -6                   	PC=0x00000020
0x00000020:	l8 r3,[r2+0]             	R3=MEM[0x00000068]=0x0A
0x00000024:	cmpi r3,0                	SR=0x00000000
0x00000028:	beq 3                    	PC=0x0000002C
0x0000002C:	s8 [r1+0],r3             	MEM[0x8888888B]=R3=0x0A
0x00000030:	addi r2,r2,1             	R2=R2+0x00000001=0x00000069,SR=0x00000000
0x00000034:	bun -6                   	PC=0x00000020
0x00000020:	l8 r3,[r2+0]             	R3=MEM[0x00000069]=0x00
0x00000024:	cmpi r3,0                	SR=0x00000040
0x00000028:	beq 3                    	PC=0x00000038
0x00000038:	ret                      	PC=MEM[0x00007FFC]=0x0000004C
0x0000004C:	int 0                    	CR=0x00000000,PC=0x00000000
[TERMINAL]

Hello world from Poxim!

[END OF SIMULATION]
